lnwallet/channel.go

```diff
@@ -6575,6 +6575,19 @@ func (lc *LightningChannel) availableCommitmentBalance(view *htlcView,
 		feePerKw.FeeForWeight(commitWeight + input.HTLCWeight),
 	)
 
+	// We want to make sure we won't try to add an HTLC that will violate
+	// the fee siphoning invariant. So if we are adding an HTLC on our
+	// commitment, the HTLC timeout fee shouldn't take the total fees that
+	// can be siphoned above our reserve.
+	if lc.channelState.ChanType.HasAnchors() && !remoteChain {
+		htlcTimeoutFee := HtlcTimeoutFee(
+			lc.channelState.ChanType, feePerKw,
+		)
+		if feeSiphon+htlcTimeoutFee > lc.channelState.LocalChanCfg.ChanReserve {
+			return 0, commitWeight
```

Comment:
Ahh, this is nice in that it'll just show up as us having no bandwidth to service a new payment from the PoV of HTLC forwarding. However users may get confused when their payment fails even though the posted channel balance should allow it....🤔